libvast/src/expression.cpp

```diff
@@ -206,9 +206,8 @@ caf::expected<expression> tailor(const expression& expr, const type& t) {
   auto x = caf::visit(type_resolver{t}, expr);
   if (!x)
     return x.error();
-  *x = caf::visit(type_pruner{t}, *x);
-  VAST_ASSERT(!caf::holds_alternative<caf::none_t>(*x));
-  return std::move(*x);
+  auto result = caf::visit(type_pruner{t}, *x);
```

Comment:
I thought about inserting

```
  if (caf::holds_alternative<caf::none_t>(*x))
    return make_error(ec::unspecified, "could not resolve type");
```

but its generating a lot of warning-level noise, and I'm also not sure that this is actually an error.